website/server/controllers/api-v3/chat.js

```diff
@@ -82,6 +84,21 @@ api.getChat = {
   },
 };
 
+function matchExact (r, str) {
+  let match = str.match(r);
+  return match !== null && match[0] !== null;
+}
+
+function textContainsBannedWords (message) {
+  for (let i = 0; i < bannedWords.length; i += 1) {
+    let word = bannedWords[i];
+    let regEx = new RegExp(`\\b${word.toLowerCase()}\\b`);
```

Comment:
since the regexp(s) are always the same for each call can we compile them only once? doing bannedWords = bannedWords.map(w => new RegRxp...)